### PR TITLE
Add advping and bossping

### PR DESCRIPTION
**Pull request by AriusX7**
At the moment, the role ping commands simply ping the role without any other message. If you have suggestions for messages, please leave them here or on Discord.

### CODE REVIEW

File: adventure/role.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+                log.exception(
+                    _("I don't have the permission to edit role permissions in {guild}.").format(
+                        guild=role.guild.name
+                    )
+                )
+
+    async def add_ping_role(self, ctx: Context, role: discord.Role, duration: Optional[str], role_type: str):
+        async def add_role():
+            try:
+                await ctx.author.add_roles(*[role])
```
Comment (fourjr) [inline adventure/role.py]:
```suggestion
                await ctx.author.add_roles(role)
```

File: adventure/role.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        self.config: Config
+
+    @commands.group(name="roleset")
+    @commands.guild_only()
+    @commands.admin_or_permissions(manage_guild=True)
+    async def _roleset(self, ctx: Context):
+        """Set roles for adventure pings."""
+
+    @_roleset.command(name="general")
+    async def _roleset_general(self, ctx: Context, *, role: Optional[discord.Role]):
```
Comment (fourjr) [inline adventure/role.py]:
```suggestion
    async def _roleset_general(self, ctx: Context, *, role: discord.Role=None):
```
It's not necessary to put optional & none is clearer.

File: adventure/role.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+            await smart_embed(ctx, _("Unset all adventures role."), True)
+        else:
+            await smart_embed(
+                ctx,
+                _("Set {role} as all adventures role.".format(role=role.mention)),
+                True
+            )
+
+    @_roleset.command(name="boss")
+    async def _roleset_boss(self, ctx: Context, *, role: Optional[discord.Role] = None):
```
Comment (fourjr) [inline adventure/role.py]:
```suggestion
    async def _roleset_boss(self, ctx: Context, *, role: discord.Role = None):
```
Not necessary to use both optional/none (similar to above)

File: adventure/role.py [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+        """Set role for all adventure pings."""
+
+        await self.config.guild(ctx.guild).general_ping_role.set(getattr(role, "id", None))
+
+        if not role:
+            await smart_embed(ctx, _("Unset all adventures role."), True)
+        else:
+            await smart_embed(
+                ctx,
+                _("Set {role} as all adventures role.".format(role=role.mention)),
```
Comment (fourjr) [inline adventure/role.py]:
```suggestion
                _("Set {role} as all adventures role.").format(role=role.mention),
```
Wrong bracket placement with the translator

**Review by fourjr — CHANGES_REQUESTED** [review]
Just some quick things I noticed. Haven't looked in depth/tested yet.

Would we want a system where it toggles? instead of an add/remove command?

File: adventure/role.py [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+
+        if await self.add_ping_role(ctx, role, duration, "general") is False:
+            return await smart_embed(ctx, _("Unable to add the role for unknown reason."))
+
+        await ctx.tick()
+
+    @commands.command()
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_roles=True)
+    async def bossrole(self, ctx: Context, *, duration: Optional[str]):
```
Comment (fourjr) [inline adventure/role.py]:
```suggestion
    async def bossrole(self, ctx: Context, *, duration=None)
```
same as above

File: adventure/role.py [hunk 6]
```diff
@@ -0,0 +1,322 @@
...
+        # the replace is needed to get the UTC timestamp
+        remove_at = (datetime.utcnow() + delta).replace(tzinfo=timezone.utc).timestamp()
+
+        async with self.config.guild(ctx.guild).timed_roles() as timed_roles:
+            timed_roles[role_type][str(ctx.author.id)] = remove_at
+
+    @commands.command()
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_roles=True)
+    async def advrole(self, ctx: Context, *, duration: Optional[str]):
```
Comment (fourjr) [inline adventure/role.py]:
```suggestion
    async def advrole(self, ctx: Context, *, duration=None):
```

**Review by fourjr — CHANGES_REQUESTED** [review]
These are the rest of my comments, looks good. Don't think I will ever have the time to test this extensively until late this week. Feel free to push after you test.

**Comment by AriusX7** [comment]
> Would we want a system where it toggles? instead of an add/remove command?

Not too sure about this. Users will have to use two commands if they want to change temporary role to permanent role if we change the behaviour to toggle (first command invocation will remove the "temp" role, second will add the role as permanent).

Personally I prefer having separate commands, but I'd be fine with having toggle behaviour.

**Review by fourjr — APPROVED** [review]

**Pull request merged**